lib/vagrant-cloudstack/action/run_instance.rb

```diff
@@ -34,8 +34,15 @@ def call(env)
 
           sanitize_domain_config
 
+          @network_ids =
+            if @domain_config.network_id.nil?
+              []
+            else
+              Array(@domain_config.network_id)
+            end
+
           @zone             = CloudstackResource.new(@domain_config.zone_id, @domain_config.zone_name, 'zone')
-          @network          = CloudstackResource.new(@domain_config.network_id, @domain_config.network_name, 'network')
+          @network          = CloudstackResource.new(@network_ids.first, @domain_config.network_name, 'network')
```

Comment:
The `@network` object is used to translate `id` from/to `name`. Probably we need an array of `@network` objects which is filled by the (array or single instance) of names and/or IDs.